processing/src/main/java/io/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -80,7 +90,9 @@ public ConcurrentGrouper(
       final ObjectMapper spillMapper,
       final int concurrencyHint,
       final DefaultLimitSpec limitSpec,
-      final boolean sortHasNonGroupingFields
+      final boolean sortHasNonGroupingFields,
+      @Nullable final ListeningExecutorService grouperSorter,
```

Comment:
It looks like this will never be null in practice. If so then please remove the `@Nullable` annotation and include a non-null check. It reduces the number of possible code paths that we would need to test.

(Currently, the null case isn't tested, so I'd like to add a test for it or get rid of it -- and getting rid of it seems easiest)